game_core/core.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package Core;
+import "types.proto";
+import "provider.proto";
+
+service GameCore {
+  // provider
+  rpc Provider(stream ProviderMsg) returns (stream ProviderMsg) {}
```

Comment:
which is the flag of game over?

Reply:
the game platform is the caller of provider?

Reply:
bi-directional connection.

but,  provider connects to game.
the advantage is
1. provider is simple
2. provider could work without public ip

I understand the provider should be an independent process logically.
But it can't work independently in our design, I don't the benefit.